scenarios/erc20_multichain_factory/test/DynamicContractRecovery_test.res

```diff
@@ -218,6 +218,48 @@ describe("Dynamic contract restart resistance test", () => {
         ~message="Should have registered only the dynamic contract up to the block that was processed",
       )
 
+      {
+        //Test the preRegistration restart function getting all the dynamic contracts
+        let setRegisterPreRegistration: (
+          Types.HandlerTypes.Register.t<'a>,
+          bool,
+        ) => unit = %raw(`(register, bool)=> {
+          if (!register.eventOptions) {
+            register.eventOptions = {};
+          } 
+          register.eventOptions.shouldPreRegisterDynamicContracts=bool;
+        }`)
+
+        Types.ERC20Factory.TokenCreated.handlerRegister->setRegisterPreRegistration(true)
```

Comment:
Would calling this here not possibly affect the other tests below this? Would it not be better to isolate this into a separate small test so logic affect other test by mistake?

Reply:
It doesn't affect the test below because the handlers aren't using pre registrations. It was the fastest way I could think of writing a test. I can try improve it but since it's a global value mutating it from any test could cause problems. Maybe the best thing todo is have a reset callback that sets it back to whatever it was before.

Reply:
You can see I reset it back to false after the assertions